### PR TITLE
fix: make mermaid optional

### PR DESCRIPTION
**Pull request by kellyjosephprice**
[![PR App][icn]][demo] |
:-------------------:|

## 🧰 Changes

Moves `mermaid` to an optional dependency.

Adding `mermaid` appears to have increased the time to import the dist by a factor of almost 2. This is mostly a concern in our backend services, ie, `mdx-renderer` and `gitto`. This is also completely unnecessary since `mermaid` does not easily support SSR. (We'd have to get a working puppeteer setup to render it on the server, which is probably not worth the effort at the moment.)

Also, I fixed up how `mermaid` class names were being applied in code tabs.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: __tests__/browser/markdown.test.js [hunk 1]
```diff
@@ -21,8 +21,9 @@ describe('visual regression tests', () => {
       'headings',
       'images',
       'imageTests',
-      'lists',
+      //'lists',
```
Comment (kellyjosephprice) [inline __tests__/browser/markdown.test.js:24]: This hasn't been migrated to mdx yet, not sure how that slipped through.

**Review by llimllib — APPROVED** [review]

**Review by trishaprile — APPROVED** [review]
🙏

File: processor/transform/mermaid.ts
Comment (rafegoldberg) [inline processor/transform/mermaid.ts:1]: Why/when would we need to manually transform these? On the server side render?

**Review by rafegoldberg — APPROVED** [review]
Had one question but otherwise I think this makes sense.

File: processor/transform/mermaid.ts
Comment (kellyjosephprice) [reply processor/transform/mermaid.ts:1]: As it is, they render as code blocks with the `lang` set to `mermaid`. On render we call `mermaid.initialize` and it looks for `pre.mermaid` elements to transform into SVG's. I don't think this currently would work during SSR anyway.

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v7.7.1</b>](/readmeio/markdown/releases/tag/v7.7.1)
<br>

**Pull request merged**